docs/src/main/asciidoc/spring-security.adoc

```diff
@@ -228,8 +228,160 @@ The table below summarizes the supported annotations:
 |@Secured
 |
 
+|@PreAuthorize
+|See next section for more details
+
 |===
 
+==== @PreAuthorize
+
+Quarkus provides support for some of the most used features of Spring Security's `@PreAuthorize` annotation.
+The expressions that are supported are the following:
+
+* hasRole
+
+To test if the current user has a specific role, the `hasRole` expression can be used inside `@PreAuthorize`.
+Some examples are: `@PreAuthorize("hasRole('admin')")`, `@PreAuthorize("hasRole(@roles.USER)")` where the `roles` is a bean that could be defined like so:
+
+[source, java]
+----
+import org.springframework.stereotype.Component;
+
+@Component
+public class Roles {
+
+    public final String ADMIN = "admin";
+    public final String USER = "user";
+}
+
+----
+
+* hasAnyRole
+
+In the same fashion as `hasRole`, users can use `hasAnyRole`  to check if the logged in user has any of the specified roles. Some examples are:
+`@PreAuthorize("hasAnyRole('admin')")`, `@PreAuthorize("hasAnyRole(@roles.USER, 'view')")`
+
+* permitAll
+
+Adding `@PreAuthorize("permitAll()")` to a method will ensure that that method is accessible by any user (including anonymous users). Adding it to a class will ensure that all public methods
+of the class that are not annotated with any other Spring Security annotation will be accessible.
+
+* denyAll
+
+Adding `@PreAuthorize("denyAll()")` to a method will ensure that that method is not accessible by any user. Adding it to a class will ensure that all public methods
+of the class that are not annotated with any other Spring Security annotation will not be accessible to any user.
+
+* isAnonymous
+
+When annotating a bean method with `@PreAuthorize("isAnonymous()")` the method will only be accessible if the current user is anonymous - i.e. a non logged in user.
+
+* isAuthenticated
+
+When annotating a bean method with `@PreAuthorize("isAuthenticated()")` the method will only be accessible if the current user is a logged in user. Essentially the
+method is only unavailable for anonymous users.
+
+* #paramName == authentication.principal.username
+
+This syntax allows users to check if a parameter (or a field of the parameter) of the secured method is equal to the logged in username.
```

Comment:
Didn't you add the support for `!=` ? I have a vague recollection of it.

Reply:
Yes support was added, so indeed it should be mentioned.